### PR TITLE
Upgrade to Relay-Compiler 4.0.0

### PR DESCRIPTION
**Pull request by thiagoleitedev**
* Update relay-compiler to version 4
* Rename RelayCompilerPublic.js to index.js

It will solve these problems:
https://github.com/facebook/relay/issues/2710
https://github.com/facebook/relay/pull/2677

### CODE REVIEW

File: src/TypeScriptGenerator.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {
   TypeGenerator,
   TypeGeneratorOptions
 } from "relay-compiler";
-import * as RelayCompilerPublic from "relay-compiler/lib/RelayCompilerPublic";
+import * as RelayCompilerPublic from "relay-compiler/lib";
```
Comment (janicduplessis) [inline src/TypeScriptGenerator.ts]: Can we use the main module instead of importing `/lib` (`import * as RelayCompilerPublic from "relay-compiler/lib"`)?

**Comment by alloy** [comment]
Thanks!

There are a lot of suspect changes in the test snapshots, though. What gives?

File: src/TypeScriptGenerator.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import {
   TypeGenerator,
   TypeGeneratorOptions
```
Comment (tyao1) [inline src/TypeScriptGenerator.ts]: I think we can just import IRVisitor, SchemaUtils here and get rid of RelayCompilerPublic.

**Comment by sibelius** [comment]
I've made a patch package for this https://gist.github.com/sibelius/a9c19e9e85b01dfc9ce5bc4a47561131

until this lands

**Comment by sibelius** [comment]
fixed here https://github.com/relay-tools/relay-compiler-language-typescript/pull/112

tks

**Pull request closed without merging**